Translated Javascript and Markup chapter to Italian (#1958)
HTTPArchive/almanac.httparchive.org: src/content/it/2020/javascript.md

## Quanto JavaScript utilizziamo?

Abbiamo detto che il tag `script` è il sesto elemento HTML più utilizzato. Analizziamo un po' più a fondo per vedere quanto JavaScript effettivamente ammonta.

Il sito mediano (il 50° percentile) invia 444 KB di JavaScript quando viene caricato su un dispositivo desktop e leggermente meno (411 KB) a un dispositivo mobile.

{{ figure_markup(
  image="bytes-2020.png",
  caption="Distribuzione della quantità di kilobyte JavaScript caricati per pagina.",
  description="Grafico a barre che mostra la distribuzione dei byte JavaScript per pagina di circa il 10%. Le pagine desktop caricano costantemente più byte JavaScript rispetto alle pagine mobile. Il 10°, 25°, 50°, 75° e 90° percentile per desktop sono: 87 KB, 209 KB, 444 KB, 826 KB e 1.322 KB.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRn1IaMxnTl0jhdC-C-vC5VLN_boJfLAaOfGJ968IalK1vPc8-dz0OkVmNY0LjMxZ6BIwSRB7xtRmIE/pubchart?oid=441749673&format=interactive",
  sheets_gid="2139688512",
  sql_file="bytes_2020.sql"
) }}

> È un po' deludente che qui non ci sia un divario maggiore. Sebbene sia pericoloso fare troppe ipotesi sulla rete o sulla potenza di elaborazione in base al fatto che il dispositivo in uso sia un telefono o un desktop (o una via di mezzo), vale la pena notare che [HTTP Archive mobile tests](./methodology#webpagetest) vengono eseguiti emulando un Moto G4 e una rete 3G. In altre parole, se ci fosse del lavoro da fare per adattarsi a circostanze tutt'altro che ideali passando meno codice, questi test dovrebbero mostrarlo.

Reviewer: Should "mobile tests" be translated? Or OK to leave as is?

  sheets_gid="1145367011",
  sql_file="bytes_by_3p.sql"
) }}

{{ figure_markup(
  image="bytes-by-3p-mobile.png",
  caption="Distribuzione del numero di byte JavaScript per host per dispositivi mobile.",
  description="Grafico a barre che mostra la distribuzione dei byte JavaScript per host per dispositivi mobili. I 10, 25, 50, 75 e 90° percentile per i byte di prima parte sono: 18, 60, 134, 275 e 560 KB. Questi valori sono costantemente inferiori ai valori del desktop, ma solo di 10-30 KB. Simile al desktop, i byte di terze parti sono superiori a quelli di prima parte, sui dispositivi mobili questa differenza non è così ampia, solo fino a 328 KB al 90° percentile.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRn1IaMxnTl0jhdC-C-vC5VLN_boJfLAaOfGJ968IalK1vPc8-dz0OkVmNY0LjMxZ6BIwSRB7xtRmIE/pubchart?oid=231883913&format=interactive",
  sheets_gid="1145367011",
  sql_file="bytes_by_3p.sql"
) }}

## Come cariciamo il nostro JavaScript?

Il modo in cui cariciamo JavaScript ha un impatto significativo sull'esperienza complessiva.

> Di default, JavaScript è _parser-blocking_. In altre parole, quando il browser scopre un elemento `script`, deve sospendere l'analisi dell'HTML finché lo script non è stato scaricato, analizzato ed eseguito. È un collo di bottiglia significativo e un contributo comune alle pagine che sono lente da visualizzare.

Reviewer: Should we translate "parser blocking" or OK to leave as is?

Author: It's Ok to leave as it

{{ figure_markup(
  image="correlations.png",
  caption="Correlazioni di JavaScript su vari aspetti dell'esperienza utente.",
  description="Grafico a barre che mostra il coefficiente di correlazione di Pearson per vari aspetti dell'esperienza utente. La correlazione dei byte al punteggio delle prestazioni di Lighthouse ha un coefficiente di correlazione di -0,47. Byte e il punteggio Lighthouse accessibilità: 0,08. Byte e tempo di blocco totale (TBT): 0,55. Byte di terze parti e punteggio prestazioni Lighthouse: -0,37. Byte di terze parti e punteggio di accessibilità di Lighthouse: 0,00. Byte di terze parti e TBT: 0.48. Il numero di script asincroni per pagina e il punteggio delle prestazioni di Lighthouse: -0,19. Script asincroni e punteggio di accessibilità di Lighthouse: 0,08. Script asincroni e TBT: 0,36.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRn1IaMxnTl0jhdC-C-vC5VLN_boJfLAaOfGJ968IalK1vPc8-dz0OkVmNY0LjMxZ6BIwSRB7xtRmIE/pubchart?oid=649523941&format=interactive",
  sheets_gid="2035770638",
  sql_file="correlations.sql"
) }}

> Il grafico sopra utilizza il [coefficiente di correlazione di Pearson](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient). C'è una definizione lunga e un po' complessa di cosa significhi precisamente, ma il succo è che stiamo cercando la forza della correlazione tra due numeri diversi. Se troviamo un coefficiente di 1,00, avremmo una correlazione positiva diretta. Una correlazione di 0,00 non mostrerebbe alcuna connessione tra due numeri. Qualunque cosa al di sotto di 0,00 indica una correlazione negativa, in altre parole, quando un numero aumenta, l'altro diminuisce.

Reviewer: ```suggestion
Il grafico sopra utilizza il [coefficiente di correlazione di Pearson](https://it.wikipedia.org/wiki/Indice_di_correlazione_di_Pearson). C'è una definizione lunga e un po' complessa di cosa significhi precisamente, ma il succo è che stiamo cercando la forza della correlazione tra due numeri diversi. Se troviamo un coefficiente di 1,00, avremmo una correlazione positiva diretta. Una correlazione di 0,00 non mostrerebbe alcuna connessione tra due numeri. Qualunque cosa al di sotto di 0,00 indica una correlazione negativa, in altre parole, quando un numero aumenta, l'altro diminuisce.
```

---

> part_number: I
> chapter_number: 2

Reviewer: We removed these recently:

```suggestion
```

title: JavaScript
description: Capitolo JavaScript del Web Almanac 2020 che copre quanto JavaScript utilizziamo sul Web, compressione, librerie e framework, caricamento e mappe di origine.
authors: [tkadlec]
reviewers: [ibnesayeed, denar90]
analysts: [rviscomi, paulcalvano]
editors: [rviscomi]
translators: [chefleo]
tkadlec_bio: Tim è un consulente e formatore di prestazioni web focalizzato sulla creazione di un web che tutti possano utilizzare. È autore di High Performance Images (O'Reilly, 2016) e Implementing Responsive Design (New Riders, 2012). Scrive di tutto ciò che riguarda il Web su <a href="https://timkadlec.com/">timkadlec.com</a>. Puoi trovarlo mentre condivide i suoi pensieri in un formato più breve su Twitter <a href="https://twitter.com/tkadlec">@tkadlec</a>.
discuss: 2038
results: https://docs.google.com/spreadsheets/d/1cgXJrFH02SHPKDGD0AelaXAdB3UI7PIb5dlS0dxVtfY/

> queries: 02_JavaScript

Reviewer: ```suggestion
```

featured_stat_1: 1,897ms
featured_stat_label_1: Tempo mediano del thread principale JS su dispositivo mobile
featured_stat_2: 37.22%
featured_stat_label_2: Percentuale di JS inutilizzati su dispositivi mobili
featured_stat_3: 12.2%
featured_stat_label_3: Percentuale di script caricati in modo asincrono
---

## Introduzione

JavaScript ha fatto molta strada dalle sue umili origini come l'ultimo dei tre capisaldi del web, insieme a CSS e HTML. Oggi JavaScript ha iniziato a infiltrarsi in un ampio spettro dello stack tecnico. Non è più limitato al lato client ed è una scelta sempre più popolare per gli strumenti di compilazione e lo scripting lato server. JavaScript si sta facendo strada anche nel livello CDN grazie alle soluzioni di edge computing.

Gli sviluppatori adorano un po' di JavaScript. Secondo il capitolo Markup, l'elemento `script` è il [sesto elemento HTML più popolare](./markup) in uso (prima di elementi come `p` e `i`, tra innumerevoli altri). Spendiamo circa 14 volte più byte su HTML rispetto a HTML, l'elemento costitutivo del Web, e 6 volte più byte rispetto a CSS.

{{ figure_markup(

>   image="page-weight-per-content-type.png",

Reviewer: Changed in #1930 

```suggestion
  image="../page-weight/bytes-distribution-content-type.png",
```

  caption="Peso medio della pagina per tipo di contenuto.",
  description="Grafico a barre che mostra il peso medio della pagina per le pagine desktop e per dispositivi mobili tra immagini, JS, CSS e HTML. Le quantità mediane di byte per ogni tipo di contenuto sulle pagine per dispositivi mobili sono: 916 KB di immagini, 411 KB di JS, 62 KB di CSS e 25 KB di HTML. Le pagine desktop tendono ad avere immagini molto più pesanti (circa 1000 KB) e quantità leggermente superiori di JS (circa 450 KB).",

>   chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQlN4Clqeb8aPc63h0J58WfBxoJluSFT6kXn45JGPghw1LGU28hzabMYAATXNY5st9TtjKrr2HnbfGd/pubchart?oid=1147150650&format=interactive"

Reviewer: Changed in #1930

```suggestion
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQlN4Clqeb8aPc63h0J58WfBxoJluSFT6kXn45JGPghw1LGU28hzabMYAATXNY5st9TtjKrr2HnbfGd/pubchart?oid=1147150650&format=interactive",
  sheets_gid="https://docs.google.com/spreadsheets/d/1wG4u0LV5PT9aN-XB1hixSFtI8KIDARTOCX0sp7ZT3h0/#378779486",
  sql_file="../page-weight/bytes_per_type_2020.sql"
```

) }}

Ma niente è gratuito, e questo è particolarmente vero per JavaScript: tutto quel codice ha un costo. Approfondiamo e diamo un'occhiata più da vicino a quanto script usiamo, come lo usiamo e quali sono le conseguenze.

## Quanto JavaScript utilizziamo?

Abbiamo detto che il tag `script` è il sesto elemento HTML più utilizzato. Analizziamo un po' più a fondo per vedere quanto JavaScript effettivamente ammonta.

Il sito mediano (il 50° percentile) invia 444 KB di JavaScript quando viene caricato su un dispositivo desktop e leggermente meno (411 KB) a un dispositivo mobile.

{{ figure_markup(

>   image="page-weight-per-content-type.png",

Reviewer: Changed in #1930 

```suggestion
  image="bytes-2020.png",
```

È importante pensare in modo critico agli strumenti che utilizziamo per costruire il Web e quali sono i compromessi, quindi ha senso esaminare attentamente ciò che vediamo in uso oggi.

### Librerie

HTTP Archive utilizza [Wappalyzer](./methodology#wappalyzer) per rilevare le tecnologie in uso su una determinata pagina. Wappalazyer tiene traccia di entrambe le librerie JavaScript (pensate a queste come una raccolta di frammenti o funzioni di supporto per facilitare lo sviluppo, come jQuery) e framework JavaScript (queste sono più probabili impalcature e forniscono modelli e strutture, come React).

Le librerie popolari in uso sono sostanzialmente invariate rispetto allo scorso anno, con jQuery che continua a dominare l'utilizzo e solo una delle 21 migliori librerie è caduta (lazy.js, sostituita da DataTables). In effetti, anche le percentuali delle migliori librerie sono cambiate appena rispetto allo scorso anno.

> {# TODO(analysts): table? showing rank, library, percentage and last years rank #}
> 

Reviewer: This was removed in #1912:

```suggestion
```

  sheets_gid="1654577118",
  sql_file="frameworks_libraries.sql"
) }}

L'anno scorso, [Houssein ha ipotizzato alcuni motivi per cui il dominio di jQuery continua](../2019/javascript#open-source-libraries-and-frameworks):

> WordPress, che viene utilizzato in oltre il 30% dei siti, include jQuery di default.
> Il passaggio da jQuery a una nuova libreria lato client può richiedere tempo a seconda delle dimensioni di un'applicazione e molti siti possono essere costituiti da jQuery oltre alle nuove librerie lato client.

Entrambe sono ipotesi molto valide e sembra che la situazione non sia cambiata molto su entrambi i fronti.

In effetti, il predominio di jQuery è ulteriormente supportato se ci si ferma a considerare che, delle 10 migliori librerie, 6 di esse sono jQuery o richiedono jQuery per essere utilizzate: jQuery UI, jQuery Migrate, FancyBox, Lightbox e Slick.

### I Framework

Quando guardiamo i framework, non vediamo nemmeno un cambiamento radicale in termini di adozione nei framework principali che sono stati evidenziati lo scorso anno. Vue.js ha visto un aumento significativo e AMP è cresciuto un po', ma la maggior parte di loro sono più o meno dove erano un anno fa.

> {# TODO(analysts): Compare same frameworks from last year's chapter to this year in bar chart? #}
> 

Reviewer: This was removed in #1912 

```suggestion
```

Otteniamo un'immagine molto simile quando guardiamo il tempo del thread principale per le pagine in cui sono stati rilevati questi strumenti.

{{ figure_markup(
  image="frameworks-main-thread.png",
  caption="Il tempo mediano del thread principale per pagina dal framework JavaScript.",
  description="Grafico a barre che mostra il tempo mediano del thread principale per framework. È difficile notare qualcosa di diverso da Ember.js, il cui tempo mediano del thread principale mobile è di oltre 20.000 millisecondi (20 secondi). Il resto dei framework sono minuscoli in confronto.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRn1IaMxnTl0jhdC-C-vC5VLN_boJfLAaOfGJ968IalK1vPc8-dz0OkVmNY0LjMxZ6BIwSRB7xtRmIE/pubchart?oid=691531699&format=interactive",
  sheets_gid="160832134",
  sql_file="main_thread_time_frameworks.sql"
) }}

> Il tempo del thread principale mobile di Ember salta fuori e distorce il grafico con il tempo necessario. Tirarlo fuori rende l'immagine un po' più facile da capire.

Reviewer: There was some extra text added in #1930 - can you translate this. this line should be:

> Ember's mobile main thread time jumps out and kind of distorts the graph with how long it takes. (I spent some more time looking into this and it appears to be heavily influenced [by one particular platform using this framework inefficiently](https://timkadlec.com/remembers/2021-01-26-what-about-ember/), rather than an underlying problem with Ember itself.) Pulling it out makes the picture a bit easier to understand.

Using Google Auto-translate I get this but feel free to improve the translation:

```suggestion
Il tempo del thread principale mobile di Ember salta fuori e distorce il grafico con il tempo necessario. 
(Ho passato un po 'più di tempo a esaminare questo aspetto e sembra essere fortemente influenzato <a hreflang="en" href="https://timkadlec.com/remembers/2021-01-26-what-about-ember/">da una particolare piattaforma che utilizza questo framework in modo inefficiente</a> , piuttosto che un problema di fondo con Ember stesso.) Tirarlo fuori rende l'immagine un po' più facile da capire.
```
